src/app/js/formats/streamgraph/Streamgraph.js

```diff
@@ -308,22 +319,38 @@ class Streamgraph extends PureComponent {
             legendItemContainer
                 .append('text')
                 .attr('class', `${css(styles.legendItemText)}`)
-                .text(cutStr(element.name));
+                .text(cutStr(element.name, column_number));
 
             const legendItemTooltip = legendItemContainer
                 .append('span')
                 .attr('class', `${css(styles.legendItemTooltip)}`)
                 .text(element.name);
 
             legendItemContainer
-                .on('mouseover', (d, i) => {
+                .on('mouseover', d => {
                     legendItemTooltip.style('visibility', 'visible');
+
+                    this.streams
+                        .transition()
+                        .duration(25)
+                        .attr('opacity', (d, j) => {
+                            return j != colorNameList.length - index - 1
+                                ? 0.3
+                                : 1;
+                        });
                 })
                 .on('mousemove', (d, i) => {
```

Comment:
Idem here for `d` and `i`.

```suggestion
                .on('mousemove', () => {
```